### PR TITLE
SetupStep fixes for Models

### PR DESCRIPTION
**Pull request by evantahler**
This PR fixes a few SetupStep bugs related to Models

* When viewing the setup steps, provide a "learn more" link for creating a Model.
* Fix a bug in which creating 1 Model did not complete the relevant Setup Step
* When a Model is selected in the UI, we now use that selected model to know where to direct users to to create the first Source, Group, Schedule, and Destination.  
  * If there is no Model selected, the `href` for the Setup Step will direct the user to make a Model
  * This required moving the loading of setup steps into the browser (useEffect) so we can check the cookie for the selected model
* Setup steps can be disabled if a prerequisite isn't complete yet, disabling the CTA link.  e.g.: if there is not yet a Model or App, the CTAs for adding a Destination and Schedule are greyed out.

Screen Shot 2021-10-11 at 11 40 45 AM

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

>

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: core/src/modules/ops/setupSteps.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export namespace SetupStepOps {
         "Models define the shape of your Records.  Grouparoo uses a Profile Model by default.  You can add new Models and customize them when you begin adding Sources and Properties.",
       href: "/models",
       cta: "Add a Model",
-      helpLink: null,
+      helpLink: `${configURL}/models`,
       check: async () => {
         const count = await GrouparooModel.count();
         return count > 1;
```
Comment (teallarson) [inline core/src/modules/ops/setupSteps.ts]: Nice! Could be in a different PR, but the "check" on this setup step should be watching for count > 0 now that we don't have a default model.

**Review by teallarson — APPROVED** [review]

File: core/src/modules/ops/setupSteps.ts (hunk 1 above)
Comment (evantahler) [reply core/src/modules/ops/setupSteps.ts]: Good point - I fixed all the setupStep bugs related to Models in this on PR now

File: core/src/modules/ops/setupSteps.ts [hunk 2]
```diff
@@ -26,159 +26,170 @@ export namespace SetupStepOps {
...
+        description:
+          "Add your first Source to Grouparoo.  Sources connect to Apps and are able to import data.",
+        href: modelId ? `/model/${modelId}/sources` : "/models",
+        cta: "Add a Source",
+        helpLink: `${configURL}/sources`,
+        check: async () => {
+          const count = await Source.count();
+          return count > 0;
+        },
+        disabled: async () => (await App.count()) === 0,
```
Comment (teallarson) [inline core/src/modules/ops/setupSteps.ts:101]: I like these.  I think that provides a nice layer of clarity for users to disable steps they can't do yet.

**Review by teallarson — APPROVED** [review]
These changes look really helpful for user experience.  Nice.
